### PR TITLE
Implement `dpnp.nancumprod()` through `dpnp.cumprod` call 

### PR DESCRIPTION
**Pull request by antonwolfy**
The PR proposes to implement `dpnp.nancumprod()` through existing call of `dpnp.cumprod` function.
All the relating tests are unmuted now.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_iface_nanfunctions.py [hunk 1]
```diff
@@ -271,22 +282,26 @@ def nancumprod(x1, **kwargs):
...
-    >>> a = np.array([1., np.nan])
-    >>> result = np.nancumprod(a)
-    >>> [x for x in result]
-    [1.0, 1.0]
-    >>> b = np.array([[1., 2., np.nan], [4., np.nan, 6.]])
-    >>> result = np.nancumprod(b)
-    >>> [x for x in result]
-    [1.0, 2.0, 2.0, 8.0, 8.0, 48.0]
+    >>> np.nancumprod(np.array(1))
+    array([1])
```
Comment (vtavana) [inline dpnp/dpnp_iface_nanfunctions.py:286]:
```suggestion
    array(1)
```

File: dpnp/dpnp_iface_nanfunctions.py [hunk 2]
```diff
@@ -249,19 +239,40 @@ def nanargmin(a, axis=None, out=None, *, keepdims=False):
...
+    out : {None, dpnp.ndarray, usm_ndarray}, optional
+        Alternative output array in which to place the result. It must have the
+        same shape and buffer length as the expected output but the type will
+        be cast if necessary.
+
+    Returns
+    -------
+    out : dpnp.ndarray
+        A new array holding the result is returned unless `out` is specified as
+        :class:`dpnp.ndarray`, in which case a reference to `out` is returned.
```
Comment (vtavana) [inline dpnp/dpnp_iface_nanfunctions.py:273]: The `out` array can be an `usm_ndarray` as well, right?

File: tests/third_party/cupy/math_tests/test_sumprod.py
Comment (vtavana) [inline tests/third_party/cupy/math_tests/test_sumprod.py:1]: We can remove `pytest.skip` on line#247. With the recent changes in `dpctl`, `NumPy` and `dpnp` behave similarly for returned array data type.

File: dpnp/dpnp_iface_nanfunctions.py (hunk 2 above)
Comment (antonwolfy) [reply dpnp/dpnp_iface_nanfunctions.py:273]: No, the returned array is always `dpnp.ndarray`

**Review by vtavana — APPROVED** [review]
Thank you, @antonwolfy!

**Pull request merged**